src/overlays/actors/ovl_Dm_Zl/z_dm_zl.c

```diff
@@ -30,27 +30,246 @@ const ActorInit Dm_Zl_InitVars = {
+    if (this->blinkTimer < 3) {
+        this->eyeTextureIndexRight = this->blinkTimer;
+        this->eyeTextureIndexLeft = this->blinkTimer;
+    }
+
+    switch (this->nextEyeState) {
+        case 0:
+            if (this->blinkTimer == 0) {
+                this->blinkTimer = Rand_S16Offset(30, 30);
+            }
+            break;
+        case 1:
+            if (this->blinkTimer == 0) {
+                this->eyeTextureIndexLeft = this->eyeTextureIndexRight = ZELDA_EYE_CLOSED;
+            }
+            break;
+        case 2:
+            if (this->blinkTimer == 0) {
+                this->eyeTextureIndexLeft = ZELDA_EYE_OPEN_LOOKING_LEFT;
+                this->eyeTextureIndexRight = ZELDA_EYE_OPEN_LOOKING_RIGHT;
+            }
+            break;
+        case 3:
+            if (this->blinkTimer == 0) {
+                this->eyeTextureIndexLeft = ZELDA_EYE_OPEN_LOOKING_RIGHT;
+                this->eyeTextureIndexRight = ZELDA_EYE_OPEN_LOOKING_LEFT;
+            }
+            break;
+        case 4:
+            if (this->blinkTimer == 0) {
+                this->eyeTextureIndexLeft = this->eyeTextureIndexRight = ZELDA_EYE_WIDE;
+            }
+            break;
+        case 5:
+            if (this->blinkTimer == 0) {
+                this->eyeTextureIndexLeft = this->eyeTextureIndexRight = ZELDA_EYE_HAPPY;
+            }
+            break;
+        case 6:
+            if (this->blinkTimer >= 3) {
+                this->blinkTimer = 0;
+            }
+            break;
+    }
+
+    switch (this->nextMouthState) {
+        default:
+            this->mouthTextureIndex = ZELDA_MOUTH_NEUTRAL;
+            break;
+        case 1:
+            this->mouthTextureIndex = ZELDA_MOUTH_SMILING;
+            break;
+        case 2:
+            this->mouthTextureIndex = ZELDA_MOUTH_FROWNING;
+            break;
+        case 3:
+            this->mouthTextureIndex = ZELDA_MOUTH_OPEN;
+            break;
+    }
+
+    if (this->animationIndex == ZELDA_ANIM_PLAYING_OCARINA) {
+        // override blinking: eyes closed while playing ocarina
+        this->eyeTextureIndexLeft = this->eyeTextureIndexRight = ZELDA_EYE_CLOSED;
+    }
+}
+
+void DmZl_Update(Actor* thisx, GlobalContext* globalCtx) {
+    DmZl* this = THIS;
+
+    DmZl_UpdateFace(this);
+    SkelAnime_Update(&this->skelAnime);
+    DmZl_UpdateCutscene(this, globalCtx);
+    this->actionFunc(this, globalCtx);
+}
+
+s32 DmZl_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    return false;
+}
 
-extern UNK_TYPE D_0600DE08;
-extern UNK_TYPE D_0600E038;
+void DmZl_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    DmZl* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Zl/func_80A38190.s")
+    if (limbIndex == ZL4_LIMB_RIGHT_HAND) {
+        if ((this->animationIndex >= ZELDA_ANIM_GIVING_OCARINA_START) && (this->animationIndex < ZELDA_ANIM_MAX)) {
```

Comment:
This would be clearer with a `<=` IMO, and better for modding if someone wanted to add animations after the last one that didn't draw the Ocarina.